### PR TITLE
Ref #32: Updated signatures to allow safe to work.

### PR DESCRIPTION
**Pull request by n1k0**
Refs #32.

- [x] `KintoClient#deleteBucket`
- [x] `Bucket#deleteCollection`
- [x] `Bucket#setPermissions`
- [x] `Collection#setPermissions`
- [x] `Collection#setMetadata`
- [x] `Collection#setSchema`
- [x] `Collection#deleteRecord`
- [x] Find a way to keep methods like `setPermissions` and find a way to provide  `last_modified`
- [x] Add support for a `last_modified` option to delete methods
- [x] Add integration tests for concurrency checks

Stretch:

- [x] Add support for passing either a string as an identifier.

### CODE REVIEW

**Comment by n1k0** [comment]
This is now ready for review. Yes, that's huge, but important, so please focus on the documentation changes, describing the new behavior when using the `safe` option. Code is pretty much well tested so should reflect the changes described in the docs pretty accurately.

r=? @leplatrem @Natim @almet

File: README.md [hunk 1]
```diff
@@ -677,9 +779,20 @@ client.bucket("blog", {safe: true})
   .updateRecord(updatedRecord, {safe: false});
 ```
 
-### The `safe` option explained
+## The `safe` option explained
+
+The `safe` option can be used:
+
+- when creating a new resource, to ensure that any already existing one matching the provided id won't be overridden if it exists on the server;
```
Comment (almet) [inline README.md]: nit: "ID"

Comment (almet) [inline README.md]: s/one/record

**Comment by almet** [comment]
LGTM

**Comment by n1k0** [comment]
I'm landing this.

**Pull request merged**